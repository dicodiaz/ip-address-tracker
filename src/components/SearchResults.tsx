import { FC } from 'react';
import { Card, Container, Row } from 'react-bootstrap';
import { useAppSelector } from '../hooks/redux-hooks';
import useElementSize from '../hooks/useElementSize';
import { selectGeolocation } from '../redux/slices/geolocationSlice';
import SearchResultsItem from './SearchResultsItem';

const fallbackGeolocation = {
  ip: '',
  location: { region: '', country: '', postalCode: '', timezone: '' },
  isp: '',
};

const SearchResults: FC = () => {
  const geolocation = useAppSelector(selectGeolocation);
  const [squareRef, { height }] = useElementSize();

  const { ip, location, isp } = geolocation ?? fallbackGeolocation;
  const { region, country, postalCode, timezone } = location;

  return (
    <Container as="section" className="z-1100">
      <Card
        className="rounded-4 p-4 p-md-5"
        style={{ marginTop: -height / 2, marginBottom: -height / 2 }}
        ref={squareRef}
      >
        <Row className="mx-0 g-0 justify-content-between" xs={1} md={4}>
          <SearchResultsItem title="IP Address" text={ip} />
          <div className="d-none v-line" />
          <SearchResultsItem
            title="Location"
            text={[region, [country, postalCode].join(' ')].join(', ')}
          />
          <div className="d-none v-line" />
          <SearchResultsItem title="Timezone" text={timezone ? `UTC: ${timezone}` : ''} />
          <div className="d-none v-line" />
          <SearchResultsItem title="ISP" text={isp} />
        </Row>
      </Card>
    </Container>
  );
};

export default SearchResults;
