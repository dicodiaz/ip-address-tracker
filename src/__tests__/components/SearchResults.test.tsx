import { expect } from 'vitest';
import SearchResults from '../../components/SearchResults';
import mockGeolocation from '../../__mocks__/mockGeolocation';
import { renderWithProviders } from '../test-utils';

describe('SearchResults component', () => {
  const preloadedState = {
    geolocationReducer: { loading: false, geolocation: mockGeolocation },
  };

  it('should render as expected', async () => {
    const { container } = renderWithProviders(<SearchResults />, { preloadedState });
    expect(container).toMatchSnapshot();
  });
});
