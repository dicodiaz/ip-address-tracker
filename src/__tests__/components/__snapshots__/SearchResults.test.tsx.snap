// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchResults component > should render as expected 1`] = `
<div>
  <section
    class="z-1100 container"
  >
    <div
      class="rounded-4 p-4 p-md-5 card"
      style="margin-top: 0px; margin-bottom: 0px;"
    >
      <div
        class="mx-0 g-0 justify-content-between row row-cols-md-4 row-cols-1"
      >
        <div
          class="col"
        >
          <h6
            class="search-results-item-title"
          >
            IP Address
          </h6>
          <p
            class="search-results-item-text"
          >
            8.8.8.8
          </p>
        </div>
        <div
          class="d-none v-line"
        />
        <div
          class="col"
        >
          <h6
            class="search-results-item-title"
          >
            Location
          </h6>
          <p
            class="search-results-item-text"
          >
            California, US 94043
          </p>
        </div>
        <div
          class="d-none v-line"
        />
        <div
          class="col"
        >
          <h6
            class="search-results-item-title"
          >
            Timezone
          </h6>
          <p
            class="search-results-item-text"
          >
            UTC: -07:00
          </p>
        </div>
        <div
          class="d-none v-line"
        />
        <div
          class="col"
        >
          <h6
            class="search-results-item-title"
          >
            ISP
          </h6>
          <p
            class="search-results-item-text"
          >
            Google LLC
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
